### PR TITLE
Update the categorized renderer description

### LINKED ISSUE #3334: [FEATURE] Add option to merge categories in categorized renderer
Original commit: https://github.com/qgis/QGIS/commit/7b245d1857d76bb2cad9a25bb3cda0b2f7bcd784 by nyalldawson

This allows users to select multiple existing categories and
group them into a single category, which applies to any of the
values from the selection.

This allows simpler styling of layers with a large number of
categories, where it may be possible to group numerous distinct
categories into a smaller, more managable set of categories
which apply to multiple values.

The option is available from the right click context menu
in the categories list view, whenever multiple categories
are selected.

Sponsored by SMEC/SJ

### PR DESCRIPTION
**Pull request by DelazJ**
using a step-like structure and add some details
ref #4230, ref #4056, Fixes #3334, ref #1896

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
-    ``field_1 + field_2``.
-  * be used to transform linear values in discrete classes, e.g.:
+.. figure:: img/categorysymbol_ng_line.png
+   :align: center
 
-    ::
+   Categorized Symbolizing options
 
-     CASE WHEN x > 1000 THEN 'Big' ELSE 'Small' END
+To apply a categorized symbology to a layer:
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
To use categorized symbology to a layer:
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
 
-  * combine several discrete values in one single category, e.g.:
+#. Select the :guilabel:`Value` of classification: it can be an existing field
+   or an :ref:`expression <vector_expressions>` you can type in the box or build
+   using the associated |expression| button. Using expressions for categorizing
+   avoids the need to create ad hoc field for symbology purpose (eg, if your
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   avoids the need to create an ad hoc field for symbology purposes (eg, if your
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+   classification criteria is derived from one or more attributes).
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   classification criteria are derived from one or more attributes).
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
 
-.. _figure_categorized_symbology:
+      myfield % 2
+      year( myfield )
+      field_1 + field_2
+      substr( field_1, -3 )
 
-.. figure:: img/categorysymbol_ng_line.png
-   :align: center
+   * be used to transform linear values in discrete classes, e.g.:
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   * be used to transform linear values to discrete classes, e.g.:
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
 
-   Categorized Symbolizing options
+     ::
+
+      CASE WHEN x > 1000 THEN 'Big' ELSE 'Small' END
 
-.. _tip_change_multiple_symbols:
+   * combine several discrete values in one single category, e.g.:
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   * combine several discrete values into a single category, e.g.:
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 6]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+   .. tip:: While you can use any kind of expression to categorize features,
+    for some complex expressions it might be simpler to use :ref:`rule-based
+    rendering <rule_based_rendering>`.
 
-   In the [Advanced] menu, under the classes, you can choose one of the two
-   first actions to match symbol name to a category name in your classification.
-   *Matched to saved symbols* match category name with a symbol name from your
-   *Style Manager*. *Match to symbols from file* match category name to a
-   symbol name from an external file.
+#. Configure the :ref:`Symbol <symbol-selector>` which will be used as
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
#. Configure the :ref:`Symbol <symbol-selector>`, which will be used as
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 7]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+   base symbol for all the classes;
+#. Indicate the :ref:`Color ramp <color-ramp>`, ie the range of colors from which
+   the color applied to each symbol is selected.
+
+   Besides the common options of the :ref:`color ramp widget <color_ramp_widget>`,
+   you can apply a |unchecked| :guilabel:`Random Color Ramp` to the categories.
+   And click the :guilabel:`Shuffle Random Colors` entry to regenerate a new set
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   You can click the :guilabel:`Shuffle Random Colors` entry to regenerate a new set
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 8]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+   of random colors if not satisfactory.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   of random colors if you are not satisfied.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 9]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+#. Then click on the :guilabel:`Classify` button to create classes from the
+   distinct values of the provided field or expression.
+#. :guilabel:`Apply` the changes if the :ref:`live update <layer_styling_panel>`
+   is not in use and each feature in the map canvas will be rendered with the
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   is not in use and each feature on the map canvas will be rendered with the
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 10]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+   symbol of its class.
+
+   By default, QGIS appends an :guilabel:`all other values` class to the list.
+   While empty at the beginning, this class is used as a default class for any
+   feature not falling into the other classes (eg, when you create features
+   with new values of classification).
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   with new values for the classification field / expression).
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 11]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+
+Further tweaks can be done to the default classification:
+
+* You can |signPlus| :sup:`Add` new categories, |signMinus| :sup:`Remove` the
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
* You can |signPlus| :sup:`Add` new categories, |signMinus| :sup:`Remove`
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 12]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+  selected one(s) or :guilabel:`Delete All` of them.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
  selected categories or :guilabel:`Delete All` of them.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 13]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+* Each class can be disabled by unchecking the checkbox at the left of the
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
* A class can be disabled by unchecking the checkbox to the left of the
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 14]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+* :guilabel:`Change output Unit...` of the selected symbol(s)
+* :guilabel:`Change Width...` of the selected line symbol(s)
+* :guilabel:`Merge Categories`: Groups multiple selected categories into a single
+  one. This allows simpler styling of layers with a large number of categories,
+  where it may be possible to group numerous distinct categories into a smaller
+  and more manageable set of categories which apply to multiple values.
+
+  .. tip:: Since the symbol kept for the merged categories is the one of the
+   topmost selected category in the list, you may want to move the category
+   whose symbol you wish to reuse at the top before merging.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   whose symbol you wish to reuse to the top before merging.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 15]
```diff
@@ -326,81 +326,127 @@ Categorized Renderer
...
+The :guilabel:`Advanced` menu gives access to options to speed classification
+or fine-tune the symbols rendering:
+
+* :guilabel:`Match to saved symbols`: Using the :ref:`symbols library
+  <vector_style_manager>`, assigns to each category a symbol whose name
+  represents the classification value of the category
+* :guilabel:`Match to symbols from file...`: Provided a file with symbols,
+  assigns to each category a symbol whose name represents the classification
+  value of the category
+* :ref:`Symbol levels... <Symbols_levels>` to define order of symbols rendering.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
* :ref:`Symbol levels... <Symbols_levels>` to define the order of symbols rendering.
```

**Comment by DelazJ** [comment]
I applied most of the changes

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: My suggestion contained an error. Corrected now.

**Comment by havatv** [comment]
If you are happy with this, @DelazJ, just squash and merge.

**Comment by lajunek** [comment]
@DelazJ I think the screenshots for the new GUI (value) still need to be updated. Do you want to go on with this issue?

**Comment by DelazJ** [comment]
Yep I'll do, as well as the others in the vector properties page. Thanks for the reminder.
But to actually have this "GUI value" issue properly fixed, the training manual also needs a review. And that is not on my todo list right now

**Comment by lajunek** [comment]
@DelazJ in the meeting with the PSC there was a decision to not support the training manual anymore. So I stopped preparing a text for this ... what do you think?

**Pull request merged**